### PR TITLE
Resove issues for 3-0-stable version

### PR DESCRIPTION
**Pull request by sachinvinsol**

### CODE REVIEW

File: app/models/spree/order/checkout_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+module SpreeSplitPayments::OrderCheckoutDecorator
```
Comment (siddharth28) [inline app/models/spree/order/checkout_decorator.rb]: @sachinvinsol create a concern for this.
Also take care of file naming convention.

File: app/models/spree/order_decorator.rb [hunk 2]
```diff
@@ -37,10 +37,13 @@ def ensure_only_one_non_partial_payment_method_present_if_multiple_payments
 
   def update_params_payment_source
     if has_checkout_step?("payment") && self.payment?
-      insert_source_params
-
-      if @updating_params[:order][:payments_attributes]['0']
-        @updating_params[:order][:payments_attributes]['0'][:amount] = order_total_after_partial_payments
+      if @updating_params[:use_existing_card] == "yes"
```
Comment (siddharth28) [inline app/models/spree/order_decorator.rb]: @sachinvinsol refactor this code.

File: app/models/spree/payment_method_decorator.rb [hunk 3]
```diff
@@ -2,10 +2,9 @@
 
   def self.available_on_checkout(guest_checkout=false)
     all.select do |p|
-      p.active && 
-      (p.display_on.blank? || p.display_on == "frontend" || p.display_on == "both") && 
-      (!guest_checkout || p.guest_checkout?) && 
-      (p.environment == Rails.env || p.environment.blank?)
+      p.active &&
```
Comment (siddharth28) [inline app/models/spree/payment_method_decorator.rb]: @sachinvinsol `p` ? change to more appropriate variable name.

File: app/models/spree/order/checkout_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  def update_from_params(params, permitted_params, request_env = {})
```
Comment (siddharth28) [inline app/models/spree/order/checkout_decorator.rb]: @sachinvinsol why do we need to override this?

File: app/assets/javascripts/spree/frontend/spree-split-payments.js [hunk 5]
```diff
@@ -69,11 +70,11 @@ SplitPayments = {
       this.uncheckNonPartialPaymentMethod();
       return true;
     } else {
-      if(!this.non_partial_payment_methods.filter(':checked').length) {
+      if(this.non_partial_payment_methods.filter(':checked').length || this.use_existing_card_checkbox[0].checked) {
```
Comment (nishant-cyro) [inline app/assets/javascripts/spree/frontend/spree-split-payments.js]: @sachinvinsol @siddharth28 - Are we sure on this that there will always an element in use_existing_card_checkbox?

**Comment by nishant-cyro** [comment]
@sachinvinsol @siddharth28 - Please answer the question. Everything else seems fine to me.

**Pull request merged**